src/components/Filter/modules/ListFilter/ListFilter.tsx

```diff
@@ -47,16 +47,25 @@ const ListFilter = (
         }
     },
     getBrowserQueryUrlValue: value => value && Array.from(value),
-    getDefaultFilterValue: () => new Set(options.map(item => item.value)),
+    getDefaultFilterValue: () => listFilterOptions.initialValue ?
+        new Set([listFilterOptions.initialValue].flat()) :
+        new Set(options.map(item => item.value)),
+
     isDefaultFilterValue: value => {
-        if (value) {
+        if (value && !listFilterOptions.initialValue) {
             if (value.size === 0) {
                 return true;
             }
             return value.size === options.length;
         }
+        if (value && listFilterOptions.initialValue) {
+            const initialValueSet = new Set([listFilterOptions.initialValue].flat());
+            return value.size === initialValueSet.size;
```

Comment:
This check unfortunately isn't enough to determine if the current value is "equal to" (the exact same as) the current value. For instance... if the initial value is ["1", "2"], and the user currently selects ["3", "4"], the lengths will equal, but this isn't actually the default value. there will probably have to be some sorting and serializing as a string or some other deep comparison involved.

You could also utilize ramda's `R.equals` here I believe as well if the elements are sorted.